backend/kernelCI_app/management/commands/helpers/kcidbng_ingester.py

```diff
@@ -498,37 +338,16 @@ def process_file(filename, trees_name, spool_dir):
     return True
 
 
-def ingest_submissions_parallel(  # noqa: C901 - orchestrator with IO + threading
+def ingest_submissions_parallel(
     spool_dir: str, trees_name: dict[str, str], max_workers: int = 5
-):
+) -> None:
     """
     Ingest submissions in parallel using ThreadPoolExecutor for I/O operations
     and a single database worker thread.
     """
 
     # Get list of JSON files to process
-    json_files = [
-        f
-        for f in os.listdir(spool_dir)
-        if os.path.isfile(os.path.join(spool_dir, f)) and f.endswith(".json")
-    ]
-    if not json_files:
-        return
-
-    total_bytes = 0
-    for f in json_files:
-        try:
-            total_bytes += os.path.getsize(os.path.join(spool_dir, f))
-        except Exception:
-            pass
-
-    _out(
-        "Spool status: %d .json files queued (%.2f MB)"
-        % (
-            len(json_files),
-            total_bytes / (1024 * 1024) if total_bytes else 0.0,
-        )
-    )
+    json_files, total_bytes = get_spool_files(spool_dir)
 
```

Comment:
if not json_files: maybe? i there is no spool files, we might need to put small delay and check again